### ISSUE TITLE
git2rdata storage format?

### ISSUE THREAD

**Issue by wlandau** [ropensci/targets#861]
## Prework

* [x] Read and agree to the [code of conduct](https://ropensci.org/code-of-conduct/) and [contributing guidelines](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/ropensci/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.
* [x] Format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Proposal

@ThierryO suggested integration with [`git2rdata`](https://docs.ropensci.org/git2rdata/index.html). A new storage format might allow more efficient storage for [`gittargets`](https://github.com/ropensci/gittargets) because the data frames are text files and Git can use the diffs. I am thinking `tar_target(..., format = "git2rdata")` could be powered by [`git2rdata::write_vc()`](https://docs.ropensci.org/git2rdata/reference/write_vc.html) and [`git2rdata::read_vc()`](https://docs.ropensci.org/git2rdata/reference/read_vc.html)

My concerns are:

1. Potential scope creep on the already [long list of storage formats](https://docs.ropensci.org/targets/reference/tar_target.html#storage-formats).
2. The list of `Suggests:` packages is already quite long: https://github.com/ropensci/targets/blob/5294be93db961a492747cdcc81f85a09648b90d5/DESCRIPTION#L75-L103

I will take some time to consider if the extra format package dependency is worth this functionality.

**Comment by ThierryO** [comment]
Maybe you could do something like [aws.s3::s3read_using()](https://github.com/cloudyr/aws.s3/blob/master/R/s3read_using.R). The user specifies the function to use and must provide the function.

**Comment by wlandau** [comment]
But I doubt this will work with `repository = "aws"` or `repository = "gcp"` because the cloud storage internals in `targets` assume there is only one output file from the target.

**Comment by ThierryO** [comment]
Is that a problem? The git2rdata format is only efficiënt in combination with a git repo. It has some overhead to optimise the file for storage in a git repo. If you don't want to store the data as plain text in git, then git2rdata is not a right tool.

**Comment by wlandau** [comment]
True, it's not a problem for `git2rdata` specifically. But it may affect other custom `tar_format()`s.
